Lib/test/test_asyncio/test_tasks.py

```diff
@@ -1548,6 +1548,35 @@ def gen():
         loop.advance_time(10)
         loop.run_until_complete(asyncio.wait([a, b]))
 
+    def test_wait_with_iterator_of_tasks(self):
+
+        def gen():
+            when = yield
+            self.assertAlmostEqual(0.1, when)
+            when = yield 0
+            self.assertAlmostEqual(0.15, when)
+            yield 0.15
+
+        loop = self.new_test_loop(gen)
+
+        a = self.new_task(loop, asyncio.sleep(0.1))
+        b = self.new_task(loop, asyncio.sleep(0.15))
+
+        async def foo():
+            done, pending = await asyncio.wait(iter([b, a]))
+            self.assertEqual(done, set([a, b]))
+            self.assertEqual(pending, set())
+            return 42
+
+        res = loop.run_until_complete(self.new_task(loop, foo()))
+        self.assertEqual(res, 42)
+        self.assertAlmostEqual(0.15, loop.time())
+
+        # Doing it again should take no time and exercise a different path.
```

Comment:
@dutradda Sure, everything looks good to me except for this, which I don't understand (and I'm sure it's on me) – what different path is being taken here? Shouldn't one `run_until_complete(self.new_task(loop, foo()))` be enough to verify this patch works?

Reply:
@jstasiak I was copied the first test and change it to use an iterator.

But you are right and just one run is enough. I have removed the second run.

Reply:
Oh, ok, LGTM now then!